core/src/main/scala/com/yahoo/maha/core/RequestModel.scala

```diff
@@ -1115,6 +1119,21 @@ object RequestModel extends Logging {
     }
   }
 
+  def validateLengthForFilterValue(publicFactOption: Option[PublicFact], publicDimOption: Option[PublicDimension], filter: Filter): Boolean = {
+    val pubCol = if (publicFactOption.isDefined) publicFactOption.get.columnsByAliasMap(filter.field) else publicDimOption.get.columnsByAliasMap(filter.field)
```

Comment:
can you combine these two statements? and also check the publicDim option is not empty.